File: Inheritance/Inheritance.java
package Inheritance;

// Base Class (Parent)
class Vehicle {
    protected String brand = "Generic Vehicle";

    public void honk() {
        System.out.println("Tuut, tuut!");
    }
}

// Derived Class (Child)
class Car extends Vehicle {
    private String modelName = "Mustang";

    public void displayInfo() {
        // Accessing inherited field
        System.out.println("Brand: " + brand);
        System.out.println("Model: " + modelName);
    }
}

public class Inheritance {
    public static void main(String[] args) {
        Car myCar = new Car();

        // Calling method from Child class
        myCar.displayInfo();

        // Calling inherited method from Parent class
        myCar.honk();
    }
}

File: Inheritance/Inheritance.cpp
#include <iostream>
#include <string>

// Base Class
class Vehicle {
public:
    std::string brand = "Generic Vehicle";

    void honk() {
        std::cout << "Tuut, tuut!\n";
    }
};

// Derived Class
class Car : public Vehicle {
public:
    std::string modelName = "Mustang";
};

int main() {
    Car myCar;

    // Accessing inherited attribute
    std::cout << "Brand: " << myCar.brand << "\n";
    std::cout << "Model: " << myCar.modelName << "\n";

    // Calling inherited method
    myCar.honk();

    return 0;
}
